### ISSUE TITLE
Anyone looking into Julia 0.5-dev specific changes?

### ISSUE THREAD

**Issue by dmbates** [JuliaInterop/RCall.jl#117]
If not I will start doing so.

**Comment by dmbates** [comment]
The `db/v0.5` branch passes tests on v0.4.5 and v0.5-dev but exhibits a peculiarity in v0.5-dev.  The R process seems to forget assigned values.
```julia
julia> versioninfo()
Julia Version 0.5.0-dev+4553
Commit f4cb80b (2016-06-06 02:05 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i5-3570 CPU @ 3.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, ivybridge)

julia> R"""
       foo <- "Bar"
       ls()
       """
RCall.RObject{RCall.StrSxp}
[1] "foo"


julia> R"""
       ls()
       """
RCall.RObject{RCall.StrSxp}
character(0)
```

However, this behavior is also the case on the `master` branch so I will create a pull request for `db/v0.5` as it doesn't make things worse than they currently are.

**Comment by randy3k** [comment]
The behavior is expected, the changes was introduced at https://github.com/JuliaStats/RCall.jl/pull/110 to fix #107 and #109.

**Comment by dmbates** [comment]
Thanks.  I'll go back to using `rcall(:load, "bnc.rda")`

**Comment by simonbyrne** [comment]
One idea that has occurred to me is that the `R""` macro could evaluate in the global environment, but the substituted variables could be in a macro-specific environment.

**Comment by randy3k** [comment]
I guess I have fixed all the 0.5 warnings for now.

**Comment by randy3k** [comment]
With 6decdb9a104d89900acfe28f09ce8a60e0009e14, `R""` now evaluates the expressions in the global and saves the substituted variables in a environment variable `#JL`.
Multiple calls to `R""` will overwrite the `#JL` variable, however, the old `#JL` will not be gc'ed as long as there are some references pointing to the variables in it.
